astropy/utils/metadata.py

```diff
@@ -81,7 +176,18 @@ def merge(left, right, merge_func=concat, metadata_conflicts='warn'):
 
         else:
             try:
-                out[key] = merge_func(left[key], right[key])
+                if merge_func is None:
+                    for left_type, right_type, merge_cls in MERGE_STRATEGY_CLASSES:
+                        if not merge_cls.enabled:
```

Comment:
I was definitely worried about the global aspect of this, but didn't immediately hit upon any really nice solution.  On the one hand using a context manager to locally opt-in to different merge strategies is very safe [1].  On the other hand it would be pain if you *do* want to have some globally added strategies (e.g. working interactively or in the notebook) and don't want to repeat the context manager over and over.

So maybe we need both.  E.g. have the base class default be `enabled=False`.  If a user's derived classes explicitly set `enabled=True` then so be it.  Maybe we could name it `globally_enabled` to be more emphatic?

[1] - @embray - you would pretty much want a context manager because most of the time users are not calling `merge` directly, it happens inside a call to table `join` or `hstack`.